src/analysis/InterferenceGraph.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ * Author: doe300
+ *
+ * See the file "LICENSE" for the full license governing this code.
+ */
+
+#include "InterferenceGraph.h"
+
+#include "../Method.h"
+#include "../Profiler.h"
+#include "ControlFlowGraph.h"
+#include "DebugGraph.h"
+#include "LivenessAnalysis.h"
+
+using namespace vc4c;
+using namespace vc4c::analysis;
+
+FastSet<InterferenceNode*> InterferenceGraph::findOverfullNodes(std::size_t numNeighbors)
+{
+    FastSet<InterferenceNode*> results;
+
+    for(auto& node : nodes)
+    {
+        if(node.second.getEdgesSize() >= numNeighbors)
+            results.emplace(&node.second);
+    }
+
+    return results;
+}
+
+std::unique_ptr<InterferenceGraph> InterferenceGraph::createGraph(Method& method)
+{
+    PROFILE_START(createInterferenceGraph);
+    std::unique_ptr<InterferenceGraph> graph(new InterferenceGraph());
```

Comment:
We should use `make_unique` like
`auto graph = std::make_unique<InterferenceGraph>(InterferenceGraph());`. This can remove overhead of allocation cost of `new`.